code/datums/components/bloodysoles.dm

```diff
@@ -53,19 +53,42 @@
 	var/obj/item/parent_item = parent
 	parent_item.update_slot_icon()
 
+
+/datum/component/bloodysoles/proc/reset_bloody_shoes()
```

Comment:
bloody_shoes previously registered to every movement even when it only did anything on a normal movement when said shoes were covered in blood. in order to lower human/carbon movement cost i made it only register to the moved signal if said shoes are actually bloody